File: frontend/src/api/adminDashboard.ts
import api from "./client";

export interface SystemHealthResponse {
  cpu_percent: number;
  memory_percent: number;
  memory_used_mb: number;
  memory_total_mb: number;
  disk_percent: number;
  disk_used_gb: number;
  disk_total_gb: number;
  process_count: number;
  uptime_seconds: number;
  api_status: string;
  db_status: string;
}

export interface AuditLogItem {
  id: number;
  timestamp: string;
  actor_email: string | null;
  actor_role: string | null;
  action: string;
  resource: string | null;
  resource_id: number | null;
  severity: string;
  message: string;
}

export interface AuditLogsResponse {
  logs: AuditLogItem[];
  total: number;
}

const base = "/dashboard/admin";

export const adminDashboardApi = {
  systemHealth: () => api.get<SystemHealthResponse>(`${base}/system-health`),
  auditLogs: (limit = 50) => api.get<AuditLogsResponse>(`${base}/audit-logs`, { params: { limit } }),
};

File: backend/app/schemas/admin_dashboard.py
from datetime import datetime

from pydantic import BaseModel


class SystemHealthResponse(BaseModel):
    cpu_percent: float
    memory_percent: float
    memory_used_mb: float
    memory_total_mb: float
    disk_percent: float
    disk_used_gb: float
    disk_total_gb: float
    process_count: int
    uptime_seconds: float
    api_status: str
    db_status: str


class AuditLogItem(BaseModel):
    id: int
    timestamp: datetime
    actor_email: str | None
    actor_role: str | None
    action: str
    resource: str | None
    resource_id: int | None
    severity: str
    message: str

    class Config:
        from_attributes = True


class AuditLogsResponse(BaseModel):
    logs: list[AuditLogItem]
    total: int
